### PR TITLE
feat: upgrade pnpm

### LINKED ISSUE #11130: Repo can't be opened on GitHub codespaces
### Astro Info

```block
-
```

### If this issue only occurs in one browser, which browser is a problem?

_No response_

### Describe the Bug

The dev container fails to build because the monorepo requires an outdated version of pnpm.

### What's the expected result?

The repo can be opened on GitHub codespaces.

### Link to Minimal Reproducible Example

https://github.com/codespaces/new?hide_repo_select=true&ref=main&repo=348060227&machine=standardLinux32gb

### Participation

- [ ] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by florian-lefebvre**
## Changes

- Closes #11130
- Upgrade pnpm to latest
- Allows renovate to update pnpm

## Testing

Tests should still pass. Smoke tests are failing because docs pnpm version need to match (https://github.com/withastro/docs/pull/8394)

## Docs

N/A, no changeset

### CODE REVIEW

File: packages/astro/package.json [hunk 1]
```diff
@@ -234,7 +234,7 @@
   "engines": {
     "node": "^18.17.1 || ^20.3.0 || >=21.0.0",
     "npm": ">=9.6.5",
-    "pnpm": ">=7.1.0"
+    "pnpm": ">=9.1.2"
```
Comment (bluwy) [inline packages/astro/package.json:237]: I don't think we should change this as the `engines` field is for users installing this package, and we say that "you can only download `astro` with pnpm >=7.1.2". We don't want to restrict this to latest.

(Personally I'd remove this line altogether as I'm not sure how useful this way, but probably another topic)

File: package.json [hunk 2]
```diff
@@ -45,9 +45,9 @@
   ],
   "engines": {
     "node": "^18.17.1 || ^20.3.0 || >=21.0.0",
-    "pnpm": ">=8.6.12"
+    "pnpm": ">=9.1.2"
```
Comment (bluwy) [inline package.json:48]: We could probably remove this altogether as we don't publish this `package.json`. The `packageManager` hint below should be enough.

**Review by bluwy — COMMENTED** [review]
One of the big change with pnpm v9 is that it will validate `packageManager` matching your current running pnpm version before executing, and error if not matching. This means everyone working on this monorepo should be using corepack.

I'd probably like to hear thoughts on this if this affects anyone. Otherwise we can disable this restriction with https://pnpm.io/npmrc#engine-strict. I heard corepack can be tricky to use if you use pnpm to manage Node.js versions.

Also, we should update `CONTRIBUTING.md` about the latest pnpm version. There's a lot of logistics when upgrading pnpm hence why I disabled it in renovate 😅

**Comment by florian-lefebvre** [comment]
I'm using corepack to manage pnpm and never had any issues tbh, curious to know what issues people have been facing

File: packages/astro/package.json (hunk 1 above)
Comment (florian-lefebvre) [reply packages/astro/package.json:237]: Do you know if there's a way with renovate to prevent auto-updating pnpm for this package?

**Comment by bluwy** [comment]
I'm ok with recommending corepack, and even requiring it if others are fine with it.

**Review by bluwy — APPROVED** [review]
I forgot about this. LGTM

**Review by ematipico — APPROVED** [review]
Let's 🐑 it

**Pull request merged**